### PR TITLE
Ability to build archived content

### LINKED ISSUE #1077: Sitemap URL inclusion is dependent on _wikihistory.json
See https://github.com/mdn/yari/blob/0b61d4c7e4968a6c49d214eb7ea1cbe565ddb35b/build/cli.js#L25
What determines if something should go in the sitemaps should be if the content comes from `CONTENT_ROOT`.

### LINKED ISSUE #1078: Ability to build archived-content
When you do `yarn build` it will [only build based on `CONTENT_ROOT`](https://github.com/mdn/yari/blob/0b61d4c7e4968a6c49d214eb7ea1cbe565ddb35b/content/document.js#L235). (the `localhost:3000` previewing will JIT on archived content though)

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #1078
Part of #1312
Fixes #1077

This works:
```
export CONTENT_ARCHIVED_ROOT=~/dev/MOZILLA/MDN/archived-content/files
export CONTENT_TRANSLATED_ROOT=~/dev/MOZILLA/MDN/translated-content/files
yarn build
```
It will take about 40 minutes and the resulting `./client/build/` will be about 6.9GB. 

You can also replace `yarn build` for `yarn dev` and you'll be able to open things like:

* http://localhost:3000/fr/docs/Web/HTML/Element/area - active but translated
* http://localhost:3000/en-US/docs/XUL/Attribute/onunload - archived

Here's what's remaining:

- [x] The GitHub URL ("Edit in GitHub") is wrong but it's actually hidden for archived and translated content. 
- [x] The sitemap xml files aren't built correctly. It should be based on all the slugs built from `CONTENT_ROOT` and (optionally) `CONTENT_TRANSLATED_ROOT`. 
- [x] If it's built from CONTENT_ARCHIVE_ROOT it should include a `<meta name="robots" content="noindex, nofollow">`
- [x] When viewing an en-US doc, it should have a list of other available languages even if we don't yet bother to display this in the document head. 
- [x] When viewing a non-en-US doc, it should have a link to the English version.

### CODE REVIEW

**Comment by peterbe** [comment]
Note-to-self; to get the GitHub URL right in CI, see https://github.com/marketplace/actions/github-environment-variables-action

File: build/cli.js [hunk 1]
```diff
@@ -52,11 +30,38 @@ async function buildDocuments() {
 
   let peakHeapBytes = 0;
 
+  // This builds up a mapping from en-US slugs to
```
Comment (escattone) [inline build/cli.js:33]:
```suggestion
  // This builds up a mapping from en-US slugs to localized slugs
```

File: build/cli.js [hunk 2]
```diff
@@ -52,11 +30,38 @@ async function buildDocuments() {
...
+    // If so, add it to the map so that when we build the en-US one, we can
+    // get an index of the *other* translations available.
+    if (translation_of) {
+      if (!translationsOf.has(translation_of)) {
+        translationsOf.set(translation_of, []);
+      }
+      translationsOf.set(translation_of, [
+        ...translationsOf.get(translation_of),
+        { slug: document.metadata.slug, locale: document.metadata.locale },
+      ]);
```
Comment (escattone) [inline build/cli.js:53]:
```suggestion
      translationsOf.get(translation_of).push(
        { slug: document.metadata.slug, locale: document.metadata.locale }
      );
```

**Review by escattone — APPROVED** [review]
@peterbe I did a quick review, and found one comment nit and one suggested change. Thanks! This is a big step forward!

**Comment by karlhorky** [comment]
@peterbe will the `ssr/monkeypatched-cheerio.js` file be staying around for a while (eg. will a new version get released in the meantime before this file disappears)?

If the answer is yes, maybe [`patch-package`](https://www.npmjs.com/package/patch-package) would be worth a consideration here. It's a great way to keep track of the monkey patching to a package - it generates a `.patch` file inside a directory `patches` that looks like this:

```patch
diff --git a/node_modules/gatsby-remark-copy-linked-files/index.js b/node_modules/gatsby-remark-copy-linked-files/index.js
index ec9d9c1..6c9b9e4 100644
--- a/node_modules/gatsby-remark-copy-linked-files/index.js
+++ b/node_modules/gatsby-remark-copy-linked-files/index.js
@@ -49,7 +49,8 @@ const getDestination = (linkNode, dir) => {
     })}` !== `${dir({})}`;
     return isValidFunction ? `${dir({
       name: linkNode.name,
-      hash: linkNode.internal.contentDigest
+      hash: linkNode.internal.contentDigest,
+      relativeDirectory: linkNode.relativeDirectory,
     })}.${linkNode.extension}` : `${dir()}/${defaultDestination(linkNode)}`;
   } else if (_.isString(dir)) {
     return `${dir}/${defaultDestination(linkNode)}`;
```

**Comment by peterbe** [comment]
Thanks for the tip @karlhorky !
It's actually pretty weird that not only did we have to create that `monkeypatched-cheerio.js` but we also had to put it into two different places in the repo 🙃
But it's not really hurting us at the moment. It's pretty safe. I'm just hoping Cheerio will get around to fix it. In fact, the most proactive way would be for me to submit a PR! Gee so much to do!

**Comment by karlhorky** [comment]
hehe no worries, good luck getting it into upstream!

**Pull request merged**